src/binaryManager.ts

```diff
@@ -98,27 +86,44 @@ async function getConfiguredMirrordBinary(): Promise<string | null> {
 export async function getMirrordBinary(): Promise<string> {
     const configured = await getConfiguredMirrordBinary();
     if (configured) {
+        await vscode.window.showInformationMessage(`Using mirrord binary specified in settings: ${configured}`);
         return configured;
     }
 
-    let foundLocal = await getLocalMirrordBinary();
-    // timeout is 1s if we have alternative or 10s if we don't
-    let timeout = foundLocal ? 1000 : 10000;
-    const latestVersion = await getLatestSupportedVersion(timeout);
-
-    // See if maybe we have it installed already, in correct version.
-    const localMirrord = await getLocalMirrordBinary(latestVersion);
-    if (localMirrord) {
-        const api = new MirrordAPI(localMirrord);
-        const installedVersion = await api.getBinaryVersion();
-        if (installedVersion === latestVersion) {
-            return localMirrord;
+    const extensionMirrordPath = getExtensionMirrordPath();
+    const latestVersion = await getLatestSupportedVersion(10000);
+
+    if (autoUpdate) {
+        await downloadMirrordBinary(extensionMirrordPath, latestVersion);
```

Comment:
What about the check that was done before, i.e. that maybe the binary is already installed in the correct version?